applications/GeoMechanicsApplication/tests/cpp_tests/test_geomechanics_time_integration_scheme.cpp

```diff
@@ -0,0 +1,220 @@
+    std::vector<std::pair<std::function<void()>, std::function<bool(const Kratos::intrusive_ptr<T> rElement)>>> CreateFunctionsAndChecksCalledOnAllComponents()
+    {
+        std::vector<std::pair<std::function<void()>, std::function<bool(const Kratos::intrusive_ptr<T> rElement)>>> functions_and_checks;
+
+        CompressedMatrix A;
+        Vector Dx;
+        Vector b;
+
+        // Create functions that need to be called in the test
+        auto finalize_solution_step = [this, &A, &Dx, &b]() {
+            mScheme.FinalizeSolutionStep(GetModelPart(), A, Dx, b);
+        };
+
+        auto initialize_function = [this, &A, &Dx, &b]() {
+            mScheme.InitializeSolutionStep(GetModelPart(), A, Dx, b);
+        };
+        auto initialize_non_linear_iteration = [this, &A, &Dx, &b]() {
+            mScheme.InitializeNonLinIteration(GetModelPart(), A, Dx, b);
+        };
+
+        auto finalize_non_linear_iteration = [this, &A, &Dx, &b]() {
+            mScheme.FinalizeNonLinIteration(GetModelPart(), A, Dx, b);
+        };
+
+        // Create functions that check if the previously mentioned functions have been called
+        auto finalize_function_check = [](const Kratos::intrusive_ptr<T> rElement) {
+            return rElement->IsSolutionStepFinalized();
+        };
+
+        auto initialize_function_check = [](const Kratos::intrusive_ptr<T> rElement) {
+            return rElement->IsSolutionStepInitialized();
+        };
+
+        auto initialize_non_linear_iteration_check =
+            [](const Kratos::intrusive_ptr<T> rCondition) {
+                return rCondition->IsNonLinIterationInitialized();
+            };
+
+        auto finalize_non_linear_iteration_check = [](const Kratos::intrusive_ptr<T> rCondition) {
+            return rCondition->IsNonLinIterationFinalized();
+        };
+
+        functions_and_checks.push_back({finalize_solution_step, finalize_function_check});
+        functions_and_checks.push_back({initialize_function, initialize_function_check});
+        functions_and_checks.push_back({initialize_non_linear_iteration,
+                                        initialize_non_linear_iteration_check});
+        functions_and_checks.push_back({finalize_non_linear_iteration,
+                                        finalize_non_linear_iteration_check});
+
+        return functions_and_checks;
+    }
+
+    template <class T>
+    std::vector<std::pair<std::function<void(const Kratos::intrusive_ptr<T> Component)>, std::function<bool(const Kratos::intrusive_ptr<T> rElement)>>> CreateFunctionsAndChecksCalledOnASingleComponent()
+    {
+        std::vector<std::pair<std::function<void(const Kratos::intrusive_ptr<T> Component)>,
+                              std::function<bool(const Kratos::intrusive_ptr<T> Component)>>>
+            functions_and_checks;
+
+        Element::EquationIdVectorType equation_id_vector_type;
+        ProcessInfo info;
+
+        auto equation_id = [this, &equation_id_vector_type,
```

Comment:
I have serious doubts whether this is correct. In the lambda expression, you capture a _reference_ to a local variable (`equation_id_vector_type`), which will go out of scope as soon as function `CreateFunctionsAndChecksCalledOnASingleComponent` returns. However, the reference is still there (in the closure). So I'm afraid this becomes a _dangling_ reference. Or am I overlooking something?

If you agree that this is incorrect, then you may fix it by creating a data member that you reference from the capture list. Or, alternatively (perhaps even better), create a container in the test's function body and pass a reference to it through the parameter list of `CreateFunctionsAndChecksCalledOnASingleComponent`. Then capture that reference. The lifetimes should then match.